frame/contracts/src/tests.rs

```diff
@@ -3565,6 +3565,56 @@ fn storage_deposit_works() {
 	});
 }
 
+#[test]
+fn storage_deposit_callee_works() {
+	let (wasm_caller, _code_hash_caller) = compile_module::<Test>("call").unwrap();
+	let (wasm_callee, _code_hash_callee) = compile_module::<Test>("store").unwrap();
+	ExtBuilder::default().existential_deposit(200).build().execute_with(|| {
+		let _ = Balances::deposit_creating(&ALICE, 1_000_000);
+
+		// Create both contracts: Constructors do nothing.
+		let addr_caller = Contracts::bare_instantiate(
+			ALICE,
+			0,
+			GAS_LIMIT,
+			None,
+			Code::Upload(wasm_caller),
+			vec![],
+			vec![],
+			false,
+		)
+		.result
+		.unwrap()
+		.account_id;
+		let addr_callee = Contracts::bare_instantiate(
+			ALICE,
+			0,
+			GAS_LIMIT,
+			None,
+			Code::Upload(wasm_callee),
+			vec![],
+			vec![],
+			false,
+		)
+		.result
+		.unwrap()
+		.account_id;
+
+		assert_ok!(Contracts::call(
+			RuntimeOrigin::signed(ALICE),
+			addr_caller,
+			0,
+			GAS_LIMIT,
+			None,
+			(100u32, &addr_callee).encode()
+		));
+
+		let callee = get_contract(&addr_callee);
+		assert_eq!(test_utils::get_balance(callee.deposit_account()), 302);
+		assert_eq!(callee.total_deposit(), 302);
```

Comment:
Is 302 a relevant value in the test? Or we only need to know that `callee.total_deposit()` equals `test_utils::get_balance(callee.deposit_account())` and that those are positive numbers?

Reply:
I think it's good to be explicit here about the expected value,
this is 200 of initial existential deposit + 2 for the key and 100 for the values.
Maybe we can add it in a comment